### ISSUE TITLE
Question about integrating PHP backend (using Node.js Auth Service with SuperTokens SDK)

### ISSUE THREAD

**Issue by xNarkon** [supertokens/supertokens-node#504]
Hey,

I'm currently creating a new product using your great solution :)

Currently, my app structure looks like this:
1. Main Backend API using Symfony (PHP) - api.example.com
2. Auth Service using NestJS (SuperTokens SDK) auth.example.com - enabled sharing cookie across all subdomains: .domain.com
3. Frontend Service using React - frontend.example.com

So, as there is no SDK for PHP yet, I had to implement something on my own. I did a simple solution like I'm taking sAccessToken from cookie (as it is shared across all subdomains). Then I'm extracting JWT (I'm decoding it, then validating using JKWS from auth.example.com and using userId from JWT in my "stateless API user model" in PHP). 

So, my first question and the most crucial, is it enough in case of security? I know that there are mechanisms like session revoke etc. but take a note that currently I just need the simplest as possible (but also safe) solution. 

If not, how to do this using SDK from Node.js - this is the best way I think because I don't have to develop my "custom" SDK setup in PHP, so I will use the Node.js backend as a proxy between PHP and Core. Probably I should call the verify session endpoint from Core, but to be honest, I don't see a possibility to do this using Node.js SDK.

Thanks in advance :)

**Comment by rishabhpoddar** [comment]
> So, my first question and the most crucial, is it enough in case of security?

Yes. The steps you are taking are correct.

After JWT verification, you should check the its claims to make sure that it has the right roles (if you are using our roles feature), and that it says that the user's email is verified (if you have email verification as required mode in the nestjs backend)

**Comment by xNarkon** [comment]
Thanks for such a quick response!

I've added claims validation using Node.js as a proxy SDK (so I'm just sending the request from PHP to Node.js). Do I also have to check for session validity somehow?

**Comment by rishabhpoddar** [comment]
If you are validating in nodeSDK and then proxying to the php backend, then you don't need to check for it again in php - since the node sdk function we expose checks it.

**Comment by xNarkon** [comment]
Well, I'm just doing something like this currently -> calling this endpoint from PHP backend

<img width="989" alt="image" src="https://user-images.githubusercontent.com/9370356/222970937-ea9c8186-1e60-4714-ac42-e4b3f3fe8ee6.png">

**Comment by rishabhpoddar** [comment]
Right. This code doesn't seem to be using our verifySession / getSession function from the node SDK. You should probably switch to using that instead of regular JWT verification, since the regular JWT verification won't check for specific claims like the email verification claim.

**Comment by xNarkon** [comment]
Sure, but how?

There is no function signature that I could use freely. GetSession() requires req and res that I don't have because I'm calling this from a different context than It assumes. Also

**Comment by rishabhpoddar** [comment]
This is what i understand about your architecture: your frontend is calling the php server directly and in php you are fetching the jwt from sAccessToken and using regular jwt verification, then you should make sure that the claims for email verification, roles etc are checked in PHP (simple boolean checks). Those claims are custom to supertokens and no JWT verification lib will check for those, so you should check them yourself in php.

You may have gotten email verification claim issues when querying the node server from your frontend in which you may be using getSession / verifySession (which does claim validation checks for email verification) - but theoretically, someone could take the JWT after login (and before email verification), and send that off to the php backend, and if php does not check for the email verification claim, it will allow users to access the app even if the email is not verified.

Another place you may have gotten email verification claim issues is on the frontend - it you are using our auth-react SDK with SessionAuth wrapper, it checks for email verification claim as well and redirects to the email verification screen (based on `mode` set in emailverification.init on the frontend.

Does this make sense? Am i misunderstanding your architecture?

**Comment by xNarkon** [comment]
Not really, this is how it looks like currently.

![arch drawio](https://user-images.githubusercontent.com/9370356/223356921-e2b403a9-69f3-4af5-9e87-788c2c27e976.png)

**Comment by rishabhpoddar** [comment]
I see. In this case, when PHP calls the nestjs backend, is it calling it only for the jwks endpoint or sending the JWT in the request and nestjs is verifying it and returning a success / failure?

**Comment by xNarkon** [comment]
Firstly, it requests JWKS to decode and validate the token itself, next using the decoded token it calls back the nestjs backend to validate claims using the method posted above in one of the screenshots.
